skada/tests/test_transformers.py

```diff
@@ -89,4 +92,50 @@ def test_DomainStratifiedSubsampleTransformer(da_dataset):
 
     ypred = pipeline.predict(X_target, sample_domain=sample_domain_target)
     assert ypred.shape[0] == X_target.shape[0]
+
+    ypred = pipeline.predict(X, sample_domain=sample_domain, allow_source=True)
+    assert ypred.shape[0] == X.shape[0]
+
+
+def test_DomainAndLabelStratifiedSubsampleTransformer(da_dataset):
+    X, y, sample_domain = da_dataset.pack_train(as_sources=["s"], as_targets=["t"])
+    sample_weight = np.ones_like(y)
+
+    train_size = 10
+
+    # test size of output on fit_transform
+    transformer = DomainAndLabelStratifiedSubsampleTransformer(
+        train_size=train_size, random_state=42
+    )
+
+    X_subsampled, y_subsampled, params = transformer.fit_transform(
+        X, y, sample_domain=sample_domain, sample_weight=sample_weight
+    )
+
+    assert X_subsampled.shape == (train_size, X.shape[1])
+    assert y_subsampled.shape[0] == train_size
+    assert params["sample_domain"].shape[0] == train_size
+    assert params["sample_weight"].shape[0] == train_size
+    # check stratification
+    assert sum(params["sample_domain"] == 1) == train_size // 2
```

Comment:
you should better check the stratification, i.e. compute frequency per (domain, label)

Reply:
Done